### DISCUSSION TITLE
Is there a way to set "prevent hotlinking" for bucket files

### DISCUSSION THREAD

**Discussion by rayNip** [minio/minio#16418] [Q&A]
I have a bucket "test" for testing and my image is "http://localhost:9000/test/2.png"
I hope that only specific domain names can access our pictures, such as "test.ray.com" to allow access to pictures in my test bucket, and other domain names cannot. How should I set it up?

I refer to the configuration of aws："https://docs.aws.amazon.com/AmazonS3/latest/userguide/example-bucket-policies.html#example-bucket-policies-HTTP-HTTPS"
I use the following configuration does not seem to take effect, I do not know if there is something wrong with my configuration:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:GetObject"
            ],
            "Resource": [
                "arn:aws:s3:::test/*"
            ],
            "Condition": {
                "StringLike": {
                    "aws:Referer": [
                        "https://test.ray.com/*"
                    ]
                }
            }
        }
    ]
}
```

**Comment by harshavardhana** [comment]
For that to take affect you need your incoming referer to be set via the browser (in the request)

**Reply by klauspost** [reply]
Note that most browsers don't send HTTPS referrer on HTTP:

[HTTP RFC - 15.1.3 Encoding Sensitive Information in URI's](https://www.rfc-editor.org/rfc/rfc2616#section-15.1.3) states:

> Clients SHOULD NOT include a Referer header field in a (non-secure) HTTP request if the referring page was transferred with a secure protocol.

Your example covers this case.

**Reply by rayNip** [reply]
> Note that most browsers don't send HTTPS referrer on HTTP:
> 
> [HTTP RFC - 15.1.3 Encoding Sensitive Information in URI's](https://www.rfc-editor.org/rfc/rfc2616#section-15.1.3) states:
> 
> > Clients SHOULD NOT include a Referer header field in a (non-secure) HTTP request if the referring page was transferred with a secure protocol.
> 
> Your example covers this case.

thank you for your reply. I actually use the https environment. The URL mentioned above was written casually to explain the problem. It is not rigorous enough. I am sorry. In fact, what I want to ask is whether "StringLike" and aws:Referer can effectively prevent hotlinking. I tried it myself, but it doesn't seem to work.

**Reply by rayNip** [reply]
> For that to take affect you need your incoming referer to be set via the browser (in the request)

Maybe I didn't express it well, "prevent hotlinking" or "Anti-Stealing-Link" or "Hotlink Protection" Forgive my English level, friend.
I want to specify whether minio has a solution for "Hotlink Protection", if not, I consider using nginx to deal with it.
